lib/net/nforceif/include/lwipopts.h

```diff
@@ -43,7 +43,9 @@
 #define LWIP_IPV4 1
 #define LWIP_IPV6 1
 #define LWIP_DEBUG 1
-extern int errno;
+#define LWIP_ERRNO_INCLUDE <errno.h>
+#define LWIP_COMPAT_MUTEX_ALLOWED
+#define SYS_LIGHTWEIGHT_PROT 0
```

Comment:
http://git.savannah.nongnu.org/cgit/lwip.git/commit/?id=6e6ce4fb78a28244c3d6581ce75d3c7391d58150 made `SYS_LIGHTWEIGHT_PROT` enabled by default. This does require additional glue code however, so I disabled it. While this matches the behavior we've had so far, we should probably hook this up eventually, since it seems to be used to make functions like `netif_poll` thread-safe.

Reply:
Consider creating an issue about this, so we don't forget.